src/utils.ts

```diff
@@ -90,14 +89,23 @@ export function useMountedState(): () => boolean {
   return get;
 }
 
-const nonIsolatedModules = [
+const predefinedNonIsolatedModules = [
   'react',
   'next/router',
   'next/dist/next-server/lib/head-manager-context',
   'next/dist/next-server/lib/router-context',
   'next/dist/next-server/lib/runtime-config',
 ];
-export function executeWithFreshModules<T>(f: () => T): T {
+
+export function executeWithFreshModules<T>(
+  f: () => T,
+  userNonIsolatedModules: string[]
```

Comment:
Would it improve readability?

```suggestion
  options: {userNonIsolatedModules: string[]}
```

Reply:
Changed to object.